pkg/systemd/terminal.jsx

```diff
@@ -95,6 +120,65 @@ const _ = cockpit.gettext;
             document.cookie = cookie;
         }
 
+        forceChangeDirectory() {
+            this.setState(prevState => ({
+                channel: this.createChannel(prevState.user, cockpit.location.options.path),
+                changePathBusy: false,
+            }));
+        }
+
+        dismiss() {
+            this.setState({
+                pathError: null,
+                changePathBusy: false,
+            });
+            cockpit.location.replace("");
```

Comment:
This looks a little odd. The effect is that when I cancel a change dir request due to "busy", it doesn't go back to the previous location, but clears the path entirely. It's mostly cosmetic, as the actual terminal session stays intact.

So mostly thinking aloud, ok.

Reply:
The reasoning behind this: If we don't clear the path but the user clears the error, and then tries to open the path again, there isn't a change because there's no navigation event, so onNavigate() doesn't run.